### PR TITLE
Update to v8.0.1

### PR DESCRIPTION
**Pull request by synapticarbors**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I'm not sure why the regro bot hasn't submitted a PR yet, but I thought I'd update it to the 8* series.

### CODE REVIEW

**Comment by synapticarbors** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+{% set version = "8.0.1" %}
 
 package:
   name: setuptools_scm
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/s/setuptools_scm/setuptools_scm-{{ version }}.tar.gz
-  sha256: 6c508345a771aad7d56ebff0e70628bf2b0ec7573762be9960214730de278f27
+  url: https://pypi.io/packages/source/s/setuptools-scm/setuptools-scm-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipe/meta.yaml:8]: The URL changed from `_` to `-`, fun!

Comment (ocefpaf) [reply recipe/meta.yaml:8]: Yep.

**Comment by ocefpaf** [comment]
> We actually need `rich` in host here?

It seems to be optional now but in this version it was declared as a build dep 😱 

https://github.com/pypa/setuptools_scm/blob/77e79f493081ca258807e36cbeb1d52441df4c2a/pyproject.toml#L8

**Comment by beckermr** [comment]
ahh that makes sense - thank you!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,6 +21,10 @@ outputs:
       host:
         - python >=3.8
         - pip
+        - packaging >=20
+        - rich
```
Comment (jakirkham) [inline recipe/meta.yaml:25]: FYI we dropped `rich` in PR: https://github.com/conda-forge/setuptools_scm-feedstock/pull/71

As best we can tell [`rich` is an optional runtime dependency]( https://github.com/pypa/setuptools_scm/blob/1ac50c22500bceda22c8188782d7cf8af051452f/pyproject.toml#L57-L59 ) and [not used at build time]( https://github.com/pypa/setuptools_scm/blob/1ac50c22500bceda22c8188782d7cf8af051452f/pyproject.toml#L5-L8 ). So it doesn't appear to be needed here

Also it was presenting issues in the bot arch migration: https://github.com/regro/cf-scripts/issues/2275

Please let us know if you have any questions

**Pull request merged**